src/unit.rs

```diff
@@ -660,8 +666,21 @@ fn parse_unit_header<R: Reader>(input: &mut R) -> Result<UnitHeader<R, R::Offset
     let mut rest = input.split(unit_length)?;
 
     let version = parse_version(&mut rest)?;
-    let offset = parse_debug_abbrev_offset(&mut rest, format)?;
-    let address_size = rest.read_u8()?;
+    let offset;
+    let address_size;
+    if 2 <= version && version <= 4 {
+        offset = parse_debug_abbrev_offset(&mut rest, format)?;
+        address_size = rest.read_u8()?;
+    } else if version == 5 {
+        let unit_type = parse_compilation_unit_type(&mut rest)?;
+        if unit_type != constants::DW_UT_compile && unit_type != constants::DW_UT_type {
```

Comment:
Change this to only accept `DW_UT_compile`. Currently it will not work for `DW_UT_type`, because we only parse the extra type unit fields when it is in the `.debug_types` section, but DWARF 5 type units are in the `.debug_info` section.